### DISCUSSION TITLE
App Privacy Details

### DISCUSSION THREAD

**Discussion by camilape** [fastlane/fastlane#17467] [General] [answered]
Hi, 

As announced at WWDC20, App Store product pages will feature a new privacy information section to help users understand an app’s privacy practices. [more info here](https://developer.apple.com/support/app-privacy-on-the-app-store/)

I'm assuming this is going to be something similar to what we do with the Advertising Identifier, every time we are going to submit a new version to the App Store we will need to provide this information, but I'm not sure. I know that this is going to start by the end of this month, Do you have any integration in mind with Fastlane to do that automatically? 

I'd appreciate your answers or information about it. 😊

Thank you

**Comment by ainame** [comment]
I checked App Store Connect API's document but the new privacy info settings aren't ready on the API either yet.  Given IDFA declaration is available on the API, the privacy info settings are likely to be available on it too, I think. If so, we should be able to integrate that with `deliver` (aka `upload_to_app_store`) easily, like it's done for IDFA declaration.
https://developer.apple.com/documentation/appstoreconnectapi/advertising_identifier_idfa_declarations
https://github.com/fastlane/fastlane/blob/5d9219e34f21e6e3bb87df87a5e621f1e7e42edc/deliver/lib/deliver/submit_for_review.rb#L88-L159

Let's wait for the release. No one knows the details right now😄

**Comment by camilape** [comment]
Thank you @ainame I'm going to be checking when this is available, and I'll keep you posted on this.

**Comment by ainame** [comment]
It has been announced finally🙂 
https://developer.apple.com/news/?id=vlj9jty9

**Reply by ainame** [reply]
Connect API doesn't seem to add new properties to answer the questions yet?

https://developer.apple.com/documentation/appstoreconnectapi/app_metadata/app_info_localizations
https://developer.apple.com/documentation/appstoreconnectapi/appinfolocalizationupdaterequest/data/attributes

**Reply by camilape** [reply]
Thank you for the information. @ainame and the roles required to answer these questions are only Account Holder or Admin 😞

**Reply by rogerluan** [reply]
cc @joshdholtz this may be something you're interested in tracking. At classic Apple pace: delivered to us (poorly) on Nov 5th, and they will start to require this from everyone on Dec 8th 🤦‍♂️ 😅

**Reply by rogerluan** [reply]
@ainame I also couldn't find docs to this API 😢

**Comment by joshdholtz** [comment]
Hello! I got a branch locally for this that I need to finish up! It’s actually only going to work with the web session. I don’t believe there is a plan for it to come official API so I’m trying to figure out how this is going to be integrated into _fastlane_ 😊 

It kind of feels like it should fit with `deliver` but I’m a little bit torn if its not going to work with an API Key. I’m kind of learning towards making it its own action 🤔 Or maybe I add it to `deliver` but error out if API Key is used?

Any thoughts on this?

**Reply by camilape** [reply]
@joshdholtz 🤔  It's a good question because it has it's own section, but the weird thing is that the app needs to be with the status _Preparing for submission_ in order to let you answer the questions. So in that case, I think it should fit with `deliver`, something similar to what you guys do with the `Advertising Identifier`. 

There is another thing that maybe it's not like the best practice and it's that if this is using only the web that means that we would need to grant admin permissions to our Fastlane user because only admins and Account Holder can answer these questions 😞

**Reply by joshdholtz** [reply]
@camilape Yeah yeah, only admins and account holder can use it which almost makes it very overkill for deliver since the rest of deliver doesn’t _really_ need either of those 😔 Now that I think about that maybe it should be its own action 😱

**Reply by ainame** [reply]
> only admins and account holder can use it which almost makes it very overkill for deliver since the rest of deliver doesn’t really need either of those 

Ah, this makes sense. I think there is the point that if we begin supporting it with another action, we can deprecate that easily later when we want to merge it into `deliver` at some point. The other way around, maintaining deprecated options in a big feature like `deliver` will be a massive pain for long term😂 

In the meantime, we can also update the deliver's document about this needing Admin or Account Holder role to update and how you can supply answers without fastlane briefly? 😇

**Reply by joshdholtz** [reply]
Good point good point! I think I will go the new action route. Hoping I can put the finishing touches on that this week 💪

**Reply by rogerluan** [reply]
@joshdholtz JW, why you believe Apple won't offer an official API for this? 😮 You mean "in reasonable time" (before their own deadline in December), or just in general? 🙀

**Comment by mohsinahd** [comment]
Do we have any update on this? is it possible to add App Privacy Details using fastlane? we manage 300+ apps and if it can't be automate, it will be a big hassle to add the details manually for each of the app :(

**Reply by anh1979** [reply]
The same over here. Please keep us updated.

**Reply by dinsen** [reply]
You can follow this PR https://github.com/fastlane/fastlane/pull/17592

**Comment by rogerluan** [comment, marked as answer]
Full docs were released here 🚀 https://docs.fastlane.tools/uploading-app-privacy-details/
